### ISSUE TITLE
Bug: [no-unused-vars] <short description of the issue>

### ISSUE THREAD

**Issue by dead-claudia** [typescript-eslint/typescript-eslint#5511]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=4.7.4&sourceType=module&code=CYUwxgNghgTiAEEQBd4DMD2GBc8B2ArgLYBGIMAsAFDWYYDU9QA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
declare let foo: number

foo++
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error - it could be defined elsewhere and thus could technically be used elsewhere. In my case, I'm using it in a file that's transcluded alongside a lot of generated code, and the variable in question is in said generated code.

### Actual Result

```
'foo' is assigned a value but never used. 3:1 - 3:4
```

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.33.1` |
| `@typescript-eslint/parser` | `5.33.1` |
| `TypeScript` | `4.7.4` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by bradzacher** [comment]
This is working as intended.

https://eslint.org/docs/latest/rules/no-unused-vars#exported

**Comment by dead-claudia** [comment]
@bradzacher The variable isn't exported, because it's a pervasive global. The corresponding pure-JS version with ESLint would've been `/* global foo */`, [but with TS, that's obviously not going to work](https://typescript-eslint.io/play#ts=4.7.4&sourceType=module&code=PQKgBA5gNg9gRgQymAZjGYTALACg0wDUhQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA).

The bug report is that variables declared should be treated similarly to ESLint's `/* global ... */` as per https://eslint.org/docs/latest/rules/no-undef.

**Comment by dead-claudia** [comment]
Just to be clear, [this code](https://typescript-eslint.io/play#ts=4.7.4&sourceType=module&code=DYUwLgBAZg9jEF4IEYCwAoDsYGodA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA) most certainly should fail, and I'm not even remotely proposing otherwise.

```
let foo = 1

foo++
```

**Comment by dead-claudia** [comment]
Also, [`/* exported */` doesn't work, either](https://typescript-eslint.io/play#ts=4.7.4&sourceType=module&code=PQKgBApgHgDg9gJwC4QCZhMAsAKFRAYwBsBDBCMIiJMAMzjgC4wA7AVwFsAjCBXXenDABeMAEYgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA).

**Comment by bradzacher** [comment]
The thing you're looking at is that you're using a really weird setup. You're using block scoped variables (let/const) to declare global variables.

ESLint has no concept of this because people just don't do this on modern codebases.

With es6 modules the community has moved away from file concatenation entirely.

From the looks of it [the export comment syntax doesn't work on let/const and only on var.](https://typescript-eslint.io/play/#ts=4.7.4&sourceType=module&code=PQKgBApgHgDg9gJwC4QCZgGZzmEwBQqEAxgDYCGCEYAbpZtgFxgB2ArgLYBGEC++WHAF4wARiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)
If I had to guess this is by design for the reasons above. 

Ultimately your code style is, frankly, an abnormal outlier. You're using patterns that people don't use.

If you want your codebase to work this way - you can create a local eslint rule which sets the `.eslintUsed` property on the declared variable to true. Setting this will cause our rule to ignore it entirely.
For examples of rules that do this out of band override see rules like [`react/jsx-uses-react`](https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-uses-react.md) or [`react/jsx-uses-vars`](https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-uses-vars.md)
